File: csharp/byps/BTestAdapter.cs
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace byps
{
    public interface BTestAdapter
    {
        String[] getServerTempFiles();
        void printServerLog(String line);
        long[] getAcitveMessagesOnServer();
        void killClientConnections();
        void setServerAppVersion(int version);
    }
}

File: java/byps/src/com/wilutions/byps/BTestAdapter.java
package com.wilutions.byps;

/**
 * This interface provides functions used in the test suite.
 * An object implementing this interface is available by the {@link BWire#getTestAdapter()} function.
 */
public interface BTestAdapter {

	/**
	 * Return the list of files in the temporary directory of the server.
	 * @return
	 */
	String[] getServerTempFiles();
	
	/**
	 * Print a line into the server log file.
	 * @param line
	 */
	void printServerLog(String line);
	
	/**
	 * Return the IDs of the currently active messages on the server.
	 * @param inclLongPolls
	 * @return
	 */
	long[] getAcitveMessagesOnServer(boolean inclLongPolls);
	
	/**
	 * Kill all connections opened by wire that owns this object. 
	 */
	void killClientConnections();
	
	/**
	 * Set the servers application version.
	 * This is used to check that client and server classes can have different versions.
	 * @param version
	 */
	void setServerAppVersion(long version);
	
}
